lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocation.java

```diff
@@ -278,7 +278,7 @@ private static void assertLocation(
             assertThat(uriIncompatibilities).isEqualTo(Set.of("skipped"));
         }
 
-        assertThat(location).isEqualTo(location);
+        assertThat(location.equals(location)).isTrue();
```

Comment:
What's wrong with the previous form?

Reply:
It violates an error-prone check: https://errorprone.info/bugpattern/TruthSelfEquals

Reply:
Yes. I figured that this is about to test equals implementation. Maybe in these cases we should use a dedicated `equals` and `hashCode` testers? (i.e. https://github.com/jqno/equalsverifier)

Reply:
We have something like that in Airlift: EquivalenceTester